frankenstein/fit.py

```diff
@@ -16,49 +16,160 @@
+        "p0" : "Scale parameter for the power spectrum's inverse Gamma prior (suggested >0, <<1)",
+        "wsmooth" : "Strength of smoothing applied to the power spectrum (suggested range 10^-4 - 10^-1)",
+      }
+    }
 
-def load_uvdata(params):
-    pass
+    print("""
+     Fit a 1D radial brightness profile with Frankenstein from the 
+     terminal with `python fit -m xx`. A .json parameter file is required. The
+     default is default_parameters.json and is of the form:\n\n""",
+     json.dumps(param_descrip, indent=4))
 
 
-def perform_fit(params):
-    pass
+def parse_parameters(parameter_file):
+    import argparse
 
+    parser = argparse.ArgumentParser("Run a Frank fit, by default using parameters in default_parameters.json")
+    parser.add_argument("-p", "--parameters", default=parameter_file, type=str, help="Parameter file (.json)")
 
-def output_results(params):
-    pass
+    args = parser.parse_args()
+    model = json.load(open(args.parameters, 'r'))
+
+    return model
+
+
+def convert_units(model): # TODO: delete after rebase
+    rout = model['hyperpriors']['rout'] / rad_to_arcsec
+
+    return rout
+
+
+def load_uvdata(data_file):
+    u, v, vis, weights = np.genfromtxt(data_file).T
+
+    return u, v, vis, weights
+
+
+def determine_geometry(model, u, v, vis, weights):
+    if not model['geometry']['fit_geometry']:
+        from frank.geometry import FixedGeometry
+        geom = FixedGeometry(0., 0., 0., 0.)
+
+    else:
+        if model['geometry']['known_geometry']:
+            from frank.geometry import FixedGeometry
+
+            geom = FixedGeometry(model['geometry']['inc'], model['geometry']['pa'], 
+                                 model['geometry']['dra'], model['geometry']['ddec'])
+
+        else:
+            from frank.geometry import FitGeometryGaussian
+
+            if fit_phase_offset:
+                geom = FitGeometryGaussian()
+
+            else:
+                geom = FitGeometryGaussian(phase_centre=(model['geometry']['dra'],
+                                           model['geometry']['ddec']))
+
+    return geom
+
+
+def perform_fit(model, rout, geom):
+    from frank.radial_fitters import FrankFitter
+
+    FF = FrankFitter(rout, model['hyperpriors']['n'], geometry=geom,
+                     alpha=model['hyperpriors']['alpha'], 
+                     weights_smooth=model['hyperpriors']['wsmooth'])
+
+    sol = FF.fit(u, v, vis, weights)
+
+    return sol
+
+
+def output_results(model, u, v, vis, weights, sol):
+    if model['input_output']['save_profile_fit']:
+        np.savetxt(savedir + 'fit.txt', 
+                   np.array([sol.r, sol.mean, np.diag(sol.covariance)**.5]).T,
+                   header='r [arcsec]\tI [Jy/sr]\tI_err [Jy/sr]')
+
+    if model['input_output']['save_vis_fit']:
+        np.savetxt(savedir + 'fit_vis.txt', 
+                   np.array([ki / (2 * np.pi), GPHF.HankelTransform(ki)]).T,
```

Comment:
ki is undefined in this function. Also why is there a 2*pi?

Reply:
just a relic from copy/paste using old code version. To be updated.